spec/requests/unlinking/unlink_defendant_maat_reference_v2_spec.rb

```diff
@@ -0,0 +1,297 @@
+# frozen_string_literal: true
+
+# rubocop:disable RSpec/MultipleMemoizedHelpers
+
+RSpec.shared_examples 'invalid unlink_attempt request for CD API' do
+  before do
+    patch "/defendants/#{defendant_id}?urn=#{prosecution_case_reference_from_fixture}",
+          params:
+  end
+
+  it 'does NOT send an unlink request to CD API' do
+    expect(a_request(:patch, api_request_path)
+          .with(body: api_request_payload.to_json, headers: json_content))
+      .not_to have_been_made
+  end
+
+  it 'renders edit' do
+    expect(response).to render_template(:edit)
+  end
+
+  it 'displays error summary with other_reason_text presence error' do
+    expect(response.body).to have_tag(:div, with: { class: 'govuk-error-summary' }) do
+      with_text error_message
+    end
+  end
+end
+
+RSpec.describe 'unlink defendant maat reference', type: :request, vcr: true, stub_unlink_v2: true do
+  include RSpecHtmlMatchers
+
+  before do
+    create(:unlink_reason, code: 1, description: 'Reason not requiring text', text_required: false)
+    create(:unlink_reason, code: 7, description: 'Reason requiring text', text_required: true)
+    allow(Feature).to receive(:enabled?).with(:laa_references).and_return(true)
+  end
+
+  let(:user) { create(:user) }
+  let(:case_urn) { 'TEST12345' }
+  let(:defendant_fixture) { load_json_stub('linked/defendant_by_reference_body.json') }
+  let(:defendant_by_id_fixture) { load_json_stub('linked_defendant.json') }
+  let(:json_api_content) { { 'Content-Type' => 'application/vnd.api+json' } }
```

Comment:
I don't think json_api_content is needed anymore in this test because your webmock is handling stubbing now